Remora.Discord.Commands/Responders/InteractionResponder.cs

```diff
@@ -130,10 +137,26 @@ IOptions<TreeSearchOptions> treeSearchOptions
                 return Result.FromSuccess();
             }
 
+            var interactionData = gatewayEvent.Data.Value;
+            interactionData.UnpackInteraction(out var command, out var parameters);
+
+            Result<BoundCommandNode> findCommandResult = FindCommandNode(command, parameters);
+            if (!findCommandResult.IsSuccess)
+            {
+                return Result.FromError(findCommandResult);
+            }
+
             if (!_options.SuppressAutomaticResponses)
             {
                 // Signal Discord that we'll be handling this one asynchronously
                 var response = new InteractionResponse(InteractionCallbackType.DeferredChannelMessageWithSource);
+
+                EphemeralAttribute? ephemeralAttribute = findCommandResult.Entity.Node.FindCustomAttributeOnLocalTree<EphemeralAttribute>();
+                if (ephemeralAttribute?.IsEphemeral == true)
+                {
+                    response = response with { Data = new InteractionCallbackData(Flags: InteractionCallbackDataFlags.Ephemeral) };
```

Comment:
Is this a valid callback response? I would expect it to require at least one of `content` or `embeds`.

Reply:
I believe so. Every property on the [callback data structure](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-interaction-callback-data-structure) is marked as optional. Further, the docs don't mention anything about expecting certain fields when including callback data in your response.

From an anecdotal point of view, commands behave exactly as expected when providing callback data in this fashion.

Reply:
Interesting! Considering this works, would you mind adding a configuration property for enabling ephemeral responses globally, too?

Reply:
Sure! I've added it to the `InteractionResponderOptions`.